### PR TITLE
Remove `ElectronicHorizon` struct

### LINKED ISSUE #2906: Electronic horizon position notification should directly contain pointer to edge tree
As of #2834, the `electronicHorizonDidUpdatePosition` notification’s user info dictionary’s `treeKey` is set to an instance of the `ElectronicHorizon` struct, whose only purpose is to hold the starting `ElectronicHorizon.Edge` of the tree. This is redundant. Assuming we don’t intend to add more properties to `MapboxNavigationNative.ElectronicHorizon` in the future, we should eliminate the struct and place the starting edge directly in the user info dictionary. The extensive struct documentation comment should be split between `ElectronicHorizon.Edge`, `ElectronicHorizonOptions`, and `treeKey`. Nested types like `ElectronicHorizon.Edge` should be renamed to no longer nest underneath `ElectronicHorizon`.

/ref https://github.com/mapbox/mapbox-navigation-ios/pull/2834#discussion_r592032215
/cc @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by azarovalex**
### Description
Fixes #2906.

### Implementation
Removed the `ElectronicHorizon` struct and renamed `ElectronicHorizon.Edge` to `ElectronicHorizonEdge`.

### CODE REVIEW

File: Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonEdge.swift [hunk 1]
```diff
@@ -1,48 +1,47 @@
...
+
+ During active turn-by-turn navigation, the user-selected route and its metadata influence the path of the electronic horizon determined by `RouteController`. During passive navigation (free-driving), no route is actively selected, so `PassiveLocationDataSource` will determine the most probable path from the vehicle’s current location. You can receive notifications about changes in the current state of the electronic horizon by observing the `Notification.Name.electronicHorizonDidUpdatePosition`, `Notification.Name.electronicHorizonDidEnterRoadObject`, and `Notification.Name.electronicHorizonDidExitRoadObject` notifications.
+
+ Use a `RoadGraph` object to get an edge with a given identifier.
+ */
+public struct ElectronicHorizonEdge {
```
Comment (bamx23) [inline Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonEdge.swift:11]: Maybe we should call it `RoadGraphEdge`? We have other entities with such semantics and this one seems to fit the same idea.

Comment (azarovalex) [reply Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonEdge.swift:11]: I think the less breaking our changes are for a user, the better it will be. Renaming from `ElectronicHorizon.Edge` to `ElectronicHorizonEdge ` feels simpler for the end user IMHO.
I also considered not removing `ElectronicHorizon` at all and instead convert it to enum without cases so that it will serve as a namespace, but then decided not to go with this soulution and just remove `ElectronicHorizon` at all.

Comment (bamx23) [reply Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonEdge.swift:11]: Booth names are "breaking". I don't consider it a huge issue while the SDK is in beta. My concern is about naming itself, but I don't insist on `RoadGraphEdge`.

Comment (1ec5) [reply Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonEdge.swift:11]: Nesting the type under `RoadGraph` as `RoadGraph.Edge` would be pretty elegant.

File: Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonEdge.swift [hunk 2]
```diff
@@ -1,48 +1,47 @@
...
+
+ During active turn-by-turn navigation, the user-selected route and its metadata influence the path of the electronic horizon determined by `RouteController`. During passive navigation (free-driving), no route is actively selected, so `PassiveLocationDataSource` will determine the most probable path from the vehicle’s current location. You can receive notifications about changes in the current state of the electronic horizon by observing the `Notification.Name.electronicHorizonDidUpdatePosition`, `Notification.Name.electronicHorizonDidEnterRoadObject`, and `Notification.Name.electronicHorizonDidExitRoadObject` notifications.
+
+ Use a `RoadGraph` object to get an edge with a given identifier.
+ */
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonEdge.swift:10]: Since this is now the documentation comment for the edge type specifically, we shouldn’t bury the lede in terms of describing an edge. Let’s start off with the first paragraph of the old documentation, then follow with the rest.

File: Sources/MapboxCoreNavigation/CoreConstants.swift [hunk 3]
```diff
@@ -343,7 +343,8 @@ extension ElectronicHorizon {
         public static let positionKey: NotificationUserInfoKey = .init(rawValue: "position")
         
         /**
-         A key in the user info dictionary of a `Notification.Name.electronicHorizonDidUpdatePosition` notification. The corresponding value is an `ElectronicHorizon` at the root of a tree of edges in the routing graph. */
+         A key in the user info dictionary of a `Notification.Name.electronicHorizonDidUpdatePosition` notification. The corresponding value is an `ElectronicHorizonEdge` at the root of a tree of edges in the routing graph. This graph represents a probable path (or paths) of a vehicle within the routing graph for a certain distance in front of the vehicle, thus extending the user’s perspective beyond the “visible” horizon as the vehicle’s position and trajectory change.
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/CoreConstants.swift:346]: 👍 It’s good that we’re putting some more detail here, now that there isn’t a general “electronic horizon” type to look up.

File: Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonEdge.swift (hunk 1 above)
Comment (azarovalex) [reply Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonEdge.swift]: Renamed `ElectronicHorizonEdge` to `RoadGraph.Edge`

File: Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonEdge.swift (hunk 2 above)
Comment (azarovalex) [reply Sources/MapboxCoreNavigation/EHorizon/ElectronicHorizonEdge.swift]: Rearranged some lines in this doc.

File: Example/ViewController+FreeDrive.swift [hunk 4]
```diff
@@ -97,17 +97,17 @@ extension ViewController {
     }
     
     @objc func didUpdateElectronicHorizonPosition(_ notification: Notification) {
-        guard let horizon = notification.userInfo?[ElectronicHorizon.NotificationUserInfoKey.treeKey] as? ElectronicHorizon else {
+        guard let horizon = notification.userInfo?[RoadGraph.NotificationUserInfoKey.treeKey] as? RoadGraph.Edge else {
```
Comment (bamx23) [inline Example/ViewController+FreeDrive.swift:100]: Maybe it's better to rename `horizon` into `startEdge` or smth?

File: Sources/MapboxCoreNavigation/CoreConstants.swift [hunk 5]
```diff
@@ -308,28 +308,28 @@ public extension Notification.Name {
     /**
      Posted when the user’s position in the electronic horizon changes. This notification may be posted multiple times after `electronicHorizonDidEnterRoadObject` until the user transitions to a new electronic horizon.
      
-     The user info dictionary contains the keys `ElectronicHorizon.NotificationUserInfoKey.positionKey`, `ElectronicHorizon.NotificationUserInfoKey.treeKey`, `ElectronicHorizon.NotificationUserInfoKey.updatesMostProbablePathKey`, and `ElectronicHorizon.NotificationUserInfoKey.distancesByRoadObjectKey`.
+     The user info dictionary contains the keys `RoadGraph.Edge.NotificationUserInfoKey.positionKey`, `RoadGraph.Edge.NotificationUserInfoKey.treeKey`, `RoadGraph.Edge.NotificationUserInfoKey.updatesMostProbablePathKey`, and `RoadGraph.Edge.NotificationUserInfoKey.distancesByRoadObjectKey`.
```
Comment (bamx23) [inline Sources/MapboxCoreNavigation/CoreConstants.swift:311]: `RoadGraph.NotificationUserInfoKey`? Here and everywhere else

File: Sources/MapboxCoreNavigation/CoreConstants.swift [hunk 6]
```diff
@@ -308,28 +308,28 @@ public extension Notification.Name {
...
+     The user info dictionary contains the keys `RoadGraph.Edge.NotificationUserInfoKey.roadObjectIdentifierKey` and `RoadGraph.Edge.NotificationUserInfoKey.transitionKey`.
     */
     static let electronicHorizonDidExitRoadObject: Notification.Name = .init(rawValue: "ElectronicHorizonDidExitRoadObject")
 }
 
-extension ElectronicHorizon {
+extension RoadGraph {
     /**
-     Keys in the user info dictionaries of various notifications posted by instances of `RouteController` or `PassiveLocationDataSource` about `ElectronicHorizon`s.
+     Keys in the user info dictionaries of various notifications posted by instances of `RouteController` or `PassiveLocationDataSource` about `RoadGraph.Edge`s.
```
Comment (bamx23) [inline Sources/MapboxCoreNavigation/CoreConstants.swift:332]: "about `RoadGraph`"?

File: Sources/MapboxCoreNavigation/EHorizon/RoadGraphEdge.swift [hunk 7]
```diff
@@ -33,16 +37,16 @@ extension ElectronicHorizon {
 
         /**
          The edges to which the user could transition from this edge.
-         
+
          The most probable path may be split at some point if some of edges have a low probability difference (±0.05). For example, `outletEdges` can contain more than one edge with `level` set to 0. Currently, there is a maximum limit of one split per electronic horizon.
          */
-        public let outletEdges: [Edge]
+        public let outletEdges: [RoadGraph.Edge]
```
Comment (bamx23) [inline Sources/MapboxCoreNavigation/EHorizon/RoadGraphEdge.swift:43]: Is this change needed?

File: Sources/MapboxCoreNavigation/EHorizon/RoadGraphEdge.swift [hunk 8]
```diff
@@ -33,16 +37,16 @@ extension ElectronicHorizon {
...
 
         init(_ native: ElectronicHorizonEdge) {
             self.identifier = UInt(native.id)
             self.level = UInt(native.level)
             self.probability = native.probability
-            self.outletEdges = native.out.map(Edge.init)
+            self.outletEdges = native.out.map(RoadGraph.Edge.init)
```
Comment (bamx23) [inline Sources/MapboxCoreNavigation/EHorizon/RoadGraphEdge.swift:49]: Is this change needed?

File: Sources/MapboxCoreNavigation/EHorizon/RoadGraphPath.swift [hunk 9]
```diff
@@ -7,7 +7,7 @@ extension RoadGraph {
     /** A position along a linear object in the road graph. */
     public struct Path {
         /** The edge identifiers that fully or partially coincide with the linear object. */
-        public let edgeIdentifiers: [ElectronicHorizon.Edge.Identifier]
+        public let edgeIdentifiers: [RoadGraph.Edge.Identifier]
```
Comment (bamx23) [inline Sources/MapboxCoreNavigation/EHorizon/RoadGraphPath.swift:10]: Do we need to explicitly mention `RoadGraph` here(as we are already inside of `RoadGraph`)? Same for `Position`

File: Sources/MapboxCoreNavigation/EHorizon/RoadGraph.swift [hunk 10]
```diff
@@ -26,7 +26,7 @@ public final class RoadGraph {
      
      - returns: A line string corresponding to the given edge identifier, or `nil` if the edge is inaccessible.
      */
-    public func edgeShape(edgeIdentifier: ElectronicHorizon.Edge.Identifier) -> LineString? {
+    public func edgeShape(edgeIdentifier: RoadGraph.Edge.Identifier) -> LineString? {
```
Comment (bamx23) [inline Sources/MapboxCoreNavigation/EHorizon/RoadGraph.swift:29]: Seems that we don't need `RoadGraph.` prefix here and in the `edgeMetadata` method.

**Review by bamx23 — APPROVED** [review]
LGTM
